app/src/main/java/com/google/droidjump/GameView.java

```diff
@@ -16,16 +16,156 @@
 
 package com.google.droidjump;
 
+import static androidx.navigation.Navigation.findNavController;
+import android.annotation.SuppressLint;
 import android.content.Context;
+import android.graphics.BitmapFactory;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.view.MotionEvent;
+import android.view.SurfaceHolder;
 import android.view.SurfaceView;
 
 public class GameView extends SurfaceView implements Runnable {
 
+    private boolean isPlaying;
+    private int screenX;
+    private int screenY;
+    private int screenMargin;
+    private SurfaceHolder surfaceHolder;
+    private MainActivity activity;
+    private Droid droid;
+    private int timePoint;
+    Thread thread;
+    private int levelTimePoints;
+    private int levelSpeed;
+
     public GameView(Context context) {
         super(context);
+        isPlaying = false;
+    }
+
+    public GameView(Context context, int screenX, int screenY, boolean isPlaying) {
+        super(context);
+        timePoint = 0;
+        surfaceHolder = getHolder();
+        activity = (MainActivity) context;
+        this.screenX = screenX;
+        this.screenY = screenY;
+        this.isPlaying = isPlaying;
+
+        // Margin in px
+        screenMargin = (int) getResources().getDimension(R.dimen.fab_margin);
+        // Measuring droid jump height
+        DrawableElement palm = new DrawableElement(/* x= */0, /* y= */0,
+                /* bitmap= */BitmapFactory.decodeResource(getResources(), R.mipmap.palm));
+        int jumpHeight = palm.getHeight() + 100;
```

Comment:
What this 100 means?

Reply:
How could I describe a jump height in this case? I wanted to be sure that Droid can jump over the highest obstacle so got the height of the palm and increased that value on "100".